### ISSUE TITLE
Speech Recognition not working with Bing Speech

### ISSUE THREAD

**Issue by stephaneey** [Microsoft/BotFramework-WebChat#745]
[fiddlertraces.zip](https://github.com/Microsoft/BotFramework-WebChat/files/1405043/fiddlertraces.zip)
Hello,

I have been trying to test the speech functionality of the webchat control following exactly what is described here https://docs.microsoft.com/en-us/bot-framework/channel-connect-webchat-speech

Things work fine when using Chrome's speaker but it doesn't when using a Bing Speech Service instance, whether I used the subscription key or the token-based approach. Note that this very same Bing Speech Service works fine with the Javascript example hosted here https://htmlpreview.github.io/?https://github.com/Azure-Samples/SpeechToText-WebSockets-Javascript/blob/master/samples/browser/Sample.html

So, I guess the problem isn't the service but the consumer which is in this case the webchat. I have attached two Fiddler traces, the working one with the js example and the non-working one of the webchat. I have compared both of them and the only difference I could spot is that the webchat control adds this:

> {"dgi":{"Groups":[{"Type":"Generic","Hints":{"ReferenceGrammar":"51dd4f25-5f12-6452-8339-e77019a622d8"}}]}}

and not the other JS application. I don't know if this has any impact but other than that, I can't see any difference.

Note that the error 1011 InternalServerError is supposed to be transient but in my case it's not since I never managed to get it working.

Additional note: this failure has nothing to do with my local environment as hosting that on an Azure Web App results in the same error.

Thanks for your help.

**Comment by danmarshall** [comment]
@shahidkhuram - can you look at this?

**Comment by shahidkhuram** [comment]
Thanks for sharing the trace @stephaneey. I took a look at the logs and I do see the failure point. I've kicked off an internal investigation.

Could you please try removing the LUIS model association with you bot, saving, and then re-associating the LUIS model? See the "Speech Priming" section of this blog post for the steps to associate a LUIS model: https://blog.botframework.com/2017/06/26/Speech-To-Text/ . This will tell us if this was a data corruption or if something in your LUIS model is inducing the failure during speech recognition.

**Comment by stephaneey** [comment]
Thanks for your answer @shahidkhuram  . I had already removed speech priming as I suspected it could be the issue but it had no effect. That said, I don't know whether you changed something but it's working now.   I tried to associate it back and it keeps working...but maybe there is a delay at DirectLIne level (caching or something?). Now that it's working and that I used speech priming back, I'll test later to see if it keeps working or not.

Thanks
